### PR TITLE
Fix JindoFSx master service issue

### LINKED ISSUE #1985: [BUG] controller reports errors when disabling jindofsx master
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

```bash
2022-07-07T11:04:22.442+0800  ERROR  jindoctl.JindoRuntime  jindofsx/hcfs.go:26  failed to find the svc, it's nil  {"jindoruntime": "default/serverless-data", "error": "failed to find the svc serverless-data in default"}
github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx.(*JindoFSxEngine).GetHCFSStatus
  /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx/hcfs.go:26
github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx.(*JindoFSxEngine).UpdateCacheOfDataset.func1
  /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx/dataset.go:122
github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry.OnError.func1
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry/util.go:51
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.ConditionFunc.WithContext.func1
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:220
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtectionWithContext
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:233
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:226
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:421
github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry.OnError
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry/util.go:50
github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry.RetryOnConflict
  /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/client-go/util/retry/util.go:104
github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx.(*JindoFSxEngine).UpdateCacheOfDataset
  /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/jindofsx/dataset.go:99
github.com/fluid-cloudnative/fluid/pkg/ddc/base.(*TemplateEngine).Sync
  /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/base/syncs.go:72
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileRuntime
  /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:241
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal
  /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:139
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/jindo.(*RuntimeReconciler).Reconcile
  /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/jindo/jindoruntime_controller.go:104
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
```

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by frankleaf**
### Ⅱ. Does this pull request fix one issue?

fixes #1985 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/jindofsx/types.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ type Master struct {
 	DnsServer           string            `yaml:"dnsServer,omitempty"`
 	NameSpace           string            `yaml:"namespace,omitempty"`
 	Labels              map[string]string `yaml:"labels,omitempty"`
+	SvcCount            int               `yaml:"svccount"`
```
Comment (cheyang) [inline pkg/ddc/jindofsx/types.go:71]: How about naming it `ServiceCount` ?

Comment (frankleaf) [reply pkg/ddc/jindofsx/types.go]: ok,done

File: pkg/ddc/jindofsx/transform.go [hunk 2]
```diff
@@ -317,6 +317,13 @@ func (e *JindoFSxEngine) transformMaster(runtime *datav1alpha1.JindoRuntime, met
 	}
 	value.Master.FileStoreProperties = propertiesFileStore
 
+	// to transform disabled
+	if e.runtime.Spec.Master.Disabled {
+		value.Master.ServiceCount = 1
```
Comment (cheyang) [inline pkg/ddc/jindofsx/transform.go:322]: 如果disable同时ReplicaCount不为0，是不是就应该配置value.Master.ServiceCount = value.Master.ReplicaCount?

Comment (frankleaf) [reply pkg/ddc/jindofsx/transform.go:322]: 理论上fuseOnly模式ReplicaCount不会不为0

Comment (cheyang) [reply pkg/ddc/jindofsx/transform.go:322]: 这是为了保持语义的一致性

Comment (frankleaf) [reply pkg/ddc/jindofsx/transform.go:322]: ReplicaCount不会为0，只会为1或者3

Comment (frankleaf) [reply pkg/ddc/jindofsx/transform.go:322]: done

**Review by cheyang — APPROVED** [review]

**Pull request merged**